packages/wrangler/src/type-generation.ts

```diff
@@ -19,6 +118,20 @@ export async function generateTypes(
 		? (await getEntry({}, config, "types")).format
 		: "modules";
 
+	// Note: we infer whether the user has provided an envInterface by checking
+	//       if it is different from the default `Env` value, this works well
+	//       besides the fact that the user itself can actually provided `Env` as
+	//       an argument... we either need to do this or removing the yargs
+	//       default value for envInterface and do `envInterface ?? "Env"`,
+	//       for a better UX we chose to go with the yargs default value
+	const userProvidedEnvInterface = envInterface !== "Env";
+
+	if (userProvidedEnvInterface && entrypointFormat === "service-worker") {
+		logger.warn(
```

Comment:
Actually I think this should be a fatal error for this command.
It would be confusing for a developer to provide a the `--env-interface` argument but for it to still output `interface Env`.

Reply:
> It would be confusing for a developer to provide a the `--env-interface` argument but for it to still output `interface Env`.

The command would not output a `interface Env`, it would simply not output an interface at all.

e.g. the output would look something like this:
```ts
export {};
declare global {
	const MY_KV: KVNamespace;
	const MY_VARIABLE: string;
}
```

Making this a fatal error feels a bit extreme to me... but I'm not completely against it either
(my reasoning here was that when you provide the env interface argument you are sort of asking the command to use that _if_ it does generate an env interface (which it does not if you're using the service worker syntax))

given my comment above, are you still convinced that the command should generate a fatal error?

Reply:
Yeah I still think this should be an error. Think about it this way: `--env-interface` is not a valid option when it cannot be used.
Or to put it the other way, why would someone add this option if it is going to have no effect?

Reply:
I've changed this to error 🙂👍 

Please have another look 🙏 

(and please let me know if the error message is ok 🙏)